pkg/controller/customdomain/customdomain_utils.go

```diff
@@ -112,12 +112,16 @@ func (r *ReconcileCustomDomain) finalizeCustomDomain(reqLogger logr.Logger, inst
 	}, ingressSecret)
 	if err != nil {
 		reqLogger.Error(err, fmt.Sprintf("Failed to get %s secret", instance.Name))
-		return err
-	}
-	err = r.client.Delete(context.TODO(), ingressSecret)
-	if err != nil {
-		reqLogger.Error(err, fmt.Sprintf("Failed to delete %s secret", instance.Name))
-		return err
+	} else {
```

Comment:
Should `err != nil` return an error and then remove this else block?

Reply:
I've updated to return err if it's not the NotFound error.